src/puppetlabs/puppetdb/utils.clj

```diff
@@ -18,6 +18,20 @@
            [java.nio.charset Charset CharsetEncoder CoderResult StandardCharsets]
            [org.postgresql.util PGobject]))
 
+(defn flush-and-exit [status]
+  "Attempts to flush *out* and *err*, reporting any failures to *err*,
+  if possible, and then invokes (System/exit status)."
+  (let [out-ex (try (flush) nil (catch Exception ex ex))]
```

Comment:
Since I didn't see that documented, I thought it'd be safer to avoid depending on it.